pkg/rpc/connection_class.go

```diff
@@ -76,3 +76,19 @@ func ConnectionClassForKey(key roachpb.RKey, def ConnectionClass) ConnectionClas
 	}
 	return def
 }
+
+// ConnectionClassOrDefault returns the passed-in connection class if it is
+// known/supported by this server. Otherwise, it returns DefaultClass.
+//
+// This helper should be used when a ConnectionClass is not "trusted", such as
+// when it was received over RPC from another node. In a mixed-version state, it
+// is possible that one node supports a connection class, and another does not.
+//
+// If the simple behaviour of falling back to DefaultClass is not acceptable, a
+// version gate must be used to transition between versions correctly.
+func ConnectionClassOrDefault(c ConnectionClass) ConnectionClass {
```

Comment:
Bummer, I thought gogoproto would coerce unknown values to the zero value by default. Did you check this?

Reply:
Proto [spec](https://protobuf.dev/programming-guides/proto3/#enum) says that unknown values are left as is:

> During deserialization, unrecognized enum values will be preserved in the message, though how this is represented when the message is deserialized is language-dependent. In languages that support open enum types with values outside the range of specified symbols, such as C++ and Go, the unknown enum value is simply stored as its underlying integer representation. In languages with closed enum types such as Java, a case in the enum is used to represent an unrecognized value, and the underlying integer can be accessed with special accessors. In either case, if the message is serialized the unrecognized value will still be serialized with the message.